test/system/trader.js

```diff
@@ -74,27 +71,28 @@ class Trader {
       const currentAssets = this.getBalance().assets; //ltc
       const currentCapital = this.getBalance().capital; //btc
       if(side === 'B') {
-        if (currentCapital <= 0.01) {
+        totalCost = quantity * price;
+        if (currentCapital <= 0.01 || totalCost > currentCapital) {
           console.log('Not enough capital to initiate order', this.config.name);
           resolve(true);
         }
-
-        totalCost = quantity * price;
-        this.getLastBuyPriceAndQuantity(price, quantity, side);
-        this.updateBalance('ASSETS', currentAssets + quantity);
-        this.updateBalance('CAPITAL', currentCapital - totalCost);
-        resolve(this.createOrder(price, side, quantity));
+        else {
+          this.getLastBuyPriceAndQuantity(price, quantity, side);
+          this.updateBalance('ASSETS', currentAssets + quantity);
+          this.updateBalance('CAPITAL', currentCapital - totalCost);
+          resolve(this.createOrder(price, side, quantity));
+        }
       }
       else {
-        if (currentAssets > 0) {
+        if (currentAssets > 0 && quantity < currentAssets) {
           totalCost = quantity * price;
           this.getLastSellPriceAndQuantity(price, quantity, side);
           this.updateBalance('ASSETS', currentAssets - quantity);
           this.updateBalance('CAPITAL', currentCapital + totalCost);
           resolve(this.createOrder(price, side, quantity));
         }
         else {
-          console.log('don\'t have an assets to sell', this.config.name);
+          console.log("don't have an assets to sell", this.config.name);
```

Comment:
What scenario 3 is testing is that the server runs these checks, not the client. So the Trader class should proceed with placing the order even when it can figure out that it will fail. The important thing is that the server will run these checks and reject the order (same for side === 'B' also)